### PR TITLE
Bug/send messages group #1397

### LINKED ISSUE #1397: Two send messages instead of one are shown for one send transaction in group chat [send]
### Description

*Type*: Bug

*Summary*: All members except sender see 2 messages for 1 send transaction sent in group chat:
screenshot_20170630-233503

only sender sees 1 expected message:
img_1915

#### Expected behavior
for 1 send transaction all group members see 1 send message 
#### Actual behavior
for 1 send transaction all group members except sender see 2 send messages

### Reproduction

- Open Status
- open group chat wth 2 members
- as member1 send ether to member2
- as member2 check how many send messages are shown in group chat

### Additional Information
* Status version: 0.9.8-56-gd743691+
* Operating System: Android and iOS

### PR DESCRIPTION
**Pull request by goranjovic**
fixes #1397 

### Summary:

Message about incoming eth intended for Wallet contact was being shown in group chat for transactions made from group chat.

### Steps to test:
- Open Status
- Start a group chat with at least 3 participants (A, B and C)
- Send some eth from A to B
- Both B and C will receive two incoming transaction messages (the correct one and the one intended for Wallet contact on B's phone)
- B will not receive a message in Wallet contact

### Expected behavior

- Both B and C receive exactly one message in group chat (saying that A sent eth to B)
- Only B receives the message in Wallet contact

[comment]: # (PRs will only be accepted if squashed into single commit.)
status: ready

### CODE REVIEW

File: src/status_im/chat/handlers/receive_message.cljs [hunk 1]
```diff
@@ -64,10 +66,10 @@
   (let [ct       (if (= content-type c/content-type-command)
                    c/content-type-wallet-command
                    c/content-type-wallet-request)
-        message' (assoc message :clock-value 0
-                                :message-id (random/id)
-                                :chat-id wallet-chat-id
-                                :content-type ct)]
+        message' (dissoc (assoc message :clock-value 0
```
Comment (flexsurfer) [inline src/status_im/chat/handlers/receive_message.cljs]: maybe better to use `->` here

File: src/status_im/chat/handlers/receive_message.cljs [hunk 2]
```diff
@@ -28,13 +28,14 @@
...
-        local-clock      (messages/get-last-clock-value chat-id')
-        clock-new        (clocks/receive clock-value local-clock)]
+  (let [same-message      (messages/get-by-id message-id)
+        current-identity  (get-current-identity db)
+        chat-id'          (or group-id chat-id from)
+        exists?           (chats/exists? chat-id')
+        active?           (chats/is-active? chat-id')
+        local-clock       (messages/get-last-clock-value chat-id')
+        clock-new         (clocks/receive clock-value local-clock)
+        recipient-pub-key (-> message :content :params :bot-db :public :recipient :whisper-identity)]
```
Comment (flexsurfer) [inline src/status_im/chat/handlers/receive_message.cljs]: maybe better to use `get-in` here

File: src/status_im/chat/handlers/receive_message.cljs (hunk 1 above)
Comment (goranjovic) [reply src/status_im/chat/handlers/receive_message.cljs]: A good point, it does look better. In fact, it should be a function. That way we can unit test it, too.

File: src/status_im/chat/handlers/receive_message.cljs (hunk 2 above)
Comment (goranjovic) [reply src/status_im/chat/handlers/receive_message.cljs]: I typically use `get-in` when there's a good chance we'll have non keyword keys (looks better than `(get "key-name")` in threading).  Is there any specific reason you think of or aesthetic?

Comment (flexsurfer) [reply src/status_im/chat/handlers/receive_message.cljs]: `get-in` is more readable, so if i see `get-in` i don't need to think, i know exactly what it does, when i see thread macro i need time to understand what this macro does in that case

Comment (goranjovic) [reply src/status_im/chat/handlers/receive_message.cljs]: Fair enough

**Review by rasom — APPROVED** [review]

**Review by flexsurfer — APPROVED** [review]

**Pull request merged**